eip-4626.md

```diff
@@ -0,0 +1,158 @@
+---
+eip: 4626
+title: Yield Bearing Vault Standard
+description: A standard for yield bearing vaults.
+author: Author: Joey Santoro (@joeysantoro), t11s (@transmissions11), Jet Jadeja (@JetJadeja)
+discussions-to: https://github.com/ethereum/EIPs/pull/4626
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-12-22
+---
+
+# Yield Bearing Vault Standard
+
+## Simple Summary
+
+A standard for yield bearing vaults.
+
+## Abstract
+
+The following standard allows for the implementation of a standard API for yield bearing vaults within smart contracts. This standard provides basic functionality for depositing and withdrawing tokens and reading balances with an optional extension for tokenized vaults using ERC-20.
+
+## Motivation
+
+Yield bearing vaults have a lack of standardization leading to diverse implementation details. Some various examples include lending markets (Compound, Aave, Fuse), aggregators (Yearn, Rari Vaults, Idle), and intrinsically interest bearing tokens (xSushi). This makes integration difficult at the aggregator or plugin layer for protocols which need to conform to many standards. This forces each protocol to implement their own adapters which are error prone and waste development resources.
+
+A standard for yield bearing vaults will allow for a similar cambrian explosion to ERC-20, unlocking access to yield in a variety of applications with little specialized effort from developers.
+
+
+## Specification
+
+### Methods
+
+#### deposit
+
+`function deposit(address _to, uint256 _value) public returns (uint256 _shares)`
+
+Deposits `_value` tokens into the vault and grants ownership of them to `_to`.
+
+MAY return a pro-rata ownership `_shares` value corresponding `_value`, if not MUST return `0`.
+
+MAY represent `_shares` using internal accounting or an ERC-20 token. 
+
+If pro-rata shares ownership is implemented, the vault SHOULD implement `balanceOf`, `redeem`, `totalSupply` and `exchangeRate`.
+
+
+#### withdraw
+
+`function withdraw(address _to, uint256 _value) public returns (uint256 _shares)`
+
+Withdraws `_value` tokens from the vault and transfers them to `_to`. 
+
+MAY return a pro-rata ownership `_shares` value corresponding to `_value`,  if not MUST return `0`.
+
+
+#### totalHoldings
+`function totalHoldings() public view returns (uint256)`
```

Comment:
There ought to be some documentation here or elsewhere about whether this view should rely on external state. One 
 can imagine a naiive implementer making totalHoldings dependent on some kind of external oracle without realizing the security implications.

Reply:
The interface is unopinionated in the same way totalSupply is unopinionated for ERC20. We expect most implementations to use a shares based model. It will always be up to the integration layer to control for "safe" vaults based on their own security model

Reply:
Why `totalHoldings` and not `totalUnderlying`, `underlyingHoldings`, or `underlyingReserves`. I think this is the only instance where "Holdings" is used to refer to underlying.

Reply:
I like totalUnderlying as well. @transmissions11 @JetJadeja wdyt?